client/src/panels/panel_declarations/misc/key_concept_panels/KeyConcepts.js

```diff
@@ -0,0 +1,160 @@
+    key_concepts_width: null,
+  };
+  constructor(props) {
+    super(props);
+    this.accordionRef = React.createRef(null);
+
+    let should_pin;
+    if (has_local_storage) {
+      try {
+        should_pin = JSON.parse(
+          localStorage.getItem(`should_pin_key_concepts`)
+        );
+      } catch {
+        should_pin = null;
+      }
+    }
+
+    this.keyConceptsContainerRef = React.createRef();
+
+    this.state = {
+      sticky: false,
+      should_pin: _.isBoolean(should_pin) ? should_pin : true,
+    };
+  }
+
+  render() {
+    const { rendered_q_a_keys, subject } = this.props;
+
+    const { sticky, should_pin, key_concepts_width } = this.state;
+
+    const sticky_data = {
+      should_pin: should_pin,
+      pin_pressed: () =>
+        this.setState((prev_state) => {
+          localStorage.setItem(
+            "should_pin_key_concepts",
+            !prev_state.should_pin
+          );
+          return { should_pin: !prev_state.should_pin };
+        }),
+    };
+
+    return (
+      <div style={{ position: "relative" }} ref={this.keyConceptsContainerRef}>
+        <MediaQuery maxWidth={breakpoints.maxMediumDevice}>
+          {(matches) => (
+            <div
+              className={classNames(
+                "mrgn-bttm-md",
+                matches && "mrgn-tp-md",
+                sticky && should_pin && "sticky"
+              )}
+              style={{
+                width: key_concepts_width,
+              }}
+            >
+              <ButtonToolbar>
+                <AutoAccordion
+                  title={text_maker("some_things_to_keep_in_mind")}
+                  ref={this.accordionRef}
+                  showPin
+                  sticky_data={sticky_data}
+                >
+                  <div style={{ paddingLeft: "10px", paddingRight: "10px" }}>
+                    <KeyConceptList
+                      question_answer_pairs={_.map(rendered_q_a_keys, (key) => [
+                        <TM
+                          key={key + "_q"}
+                          k={key + "_q"}
+                          args={{ subject }}
+                        />,
+                        <TM
+                          key={key + "_a"}
+                          k={key + "_a"}
+                          args={{ subject }}
+                        />,
+                      ])}
+                    />
+                  </div>
+                </AutoAccordion>
+
+                {/* <Overlay
+                  show={is_showing_tooltip}
+                  target={this.accordionRef.current}
+                  container={this}
+                  placement="right"
+                >
+                  <Popover id="keep_in_mind_tooltip" style={{}}>
+                    {text_maker("click_text")}
+                  </Popover>
+                </Overlay> */}
+              </ButtonToolbar>
+            </div>
+          )}
+        </MediaQuery>
+      </div>
+    );
+  }
+
+  componentDidMount() {
```

Comment:
This might be nitpicky but I think it's more logical that class has constructor -> componentDidMount -> Render.

Reply:
I was always wondering why it was structured like this when the mounting order is like this
![image](https://user-images.githubusercontent.com/25855114/98131536-5e841780-1e89-11eb-9009-042c3fe25b30.png)
found here: https://reactjs.org/docs/react-component.html

I do notice that programmers generally structure componentDidMount above render so I put it back.

Reply:
If I had to come up with a reason, it'd be that
  a) it feels reasonable to group hooks together
  b) `componentDidMount` is most likely doing something that will prompt a re-render (loading data while the component is in a spinner state, making some calculations based on the location in the DOM, etc)